src/ucp/wireup/select.c

```diff
@@ -1031,8 +1037,13 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
             bw_info.md_map            |= UCS_BIT(context->tl_rscs[rsc_index].md_index);
             bw_info.local_dev_bitmap  &= ~UCS_BIT(context->tl_rscs[rsc_index].dev_index);
             bw_info.remote_dev_bitmap &= ~UCS_BIT(address_list[addr_index].dev_index);
-            break; /* do not continue searching due to we found
-                      AM lane (and there is only one lane) */
+            if (ucp_wireup_is_ep_single_lane(ep, rsc_index)) {
+                /* if AM lane is SELF or SHMEM - then do not use more lanes */
+                return UCS_OK;
```

Comment:
so it mean before the fix we could have multi-lane with SHM as well?
also, do we need similar fix for rma lanes as well?

Reply:
i think yes, i found this issue investigating hang on OMPI finalize

pick it up to 1.4?

Reply:
yes. should we fix also rma lanes?

Reply:
RMA is fixed in this PR as well